mimic/model/nova_objects.py

```diff
@@ -0,0 +1,392 @@
+        """
+        tenant_id = self.collection.tenant_id
+        server_id = self.server_id
+        return [
+            {
+                "href": absolutize_url("v2/{0}/servers/{1}"
+                                       .format(tenant_id, server_id)),
+                "rel": "self"
+            },
+            {
+                "href": absolutize_url("{0}/servers/{1}"
+                                       .format(tenant_id, server_id)),
+                "rel": "bookmark"
+            }
+        ]
+
+    def brief_json(self, absolutize_url):
+        """
+        Brief JSON-serializable version of this server, for the non-details
+        list servers request.
+        """
+        return {
+            'name': self.server_name,
+            'links': self.links_json(absolutize_url),
+            'id': self.server_id
+        }
+
+    def detail_json(self, absolutize_url):
+        """
+        Long-form JSON-serializable object representation of this server, as
+        returned by either a GET on this individual server or a member in the
+        list returned by the list-details request.
+        """
+        template = self.static_defaults.copy()
+        tenant_id = self.collection.tenant_id
+        template.update({
+            "id": self.server_id,
+            "OS-DCF:diskConfig": self.disk_config,
+            "OS-EXT-STS:vm_state": self.status,
+            "addresses": self.addresses_json(),
+            "created": seconds_to_timestamp(self.creation_time),
+            "updated": seconds_to_timestamp(self.update_time),
+            "flavor": {
+                "id": self.flavor_ref,
+                "links": [{"href": absolutize_url(
+                    "{0}/flavors/{1}".format(tenant_id,
+                                             self.flavor_ref))}],
+                "rel": "bookmark"
+            },
+            "image": {
+                "id": self.image_ref,
+                "links": [{
+                    "href": absolutize_url("{0}/images/{1}".format(
+                        tenant_id, self.flavor_ref))
+                }]
+            }
+            if self.image_ref is not None else '',
+            "links": self.links_json(absolutize_url),
+            "metadata": self.metadata,
+            "name": self.server_name,
+            "tenant_id": tenant_id,
+            "status": self.status
+        })
+        return template
+
+    def creation_response_json(self, absolutize_url):
+        """
+        A JSON-serializable object returned for the initial creation of this
+        server.
+        """
+        return {
+            'server': {
+                "OS-DCF:diskConfig": self.disk_config,
+                "id": self.server_id,
+                "links": self.links_json(absolutize_url),
+                "adminPass": self.admin_password,
+            }
+        }
+
+    @classmethod
+    def from_creation_request_json(cls, collection, creation_json,
```

Comment:
`from_<x>` is the naming convention for a class-method constructor, so I'm following that.

The problem with `create_...` is that, as a verb, it implies that it's going through the full process of "creating" the server, which also involves *adding* it to the collection; whereas, this method is just instantiating the Server object, which may or may not be fully "created" as the caller sees fit.

This is really hair-splitting of course, but the existing convention of `from_` seems to avoid the issue of whether "create" and "instantiate" are really synonyms in this context.